bxsolana/provider/ws.ts

```diff
@@ -62,233 +62,305 @@ import {
+    cancelGetOrderbooksStreamByCount = async (streamNumber: number): Promise<boolean> => {
+        return this.cancelStreamByCount("GetOrderbooksStream", streamNumber)
     }
 
-    postCancelAll(request: PostCancelAllRequest): Promise<PostCancelAllResponse> {
-        return this.wsSocketCall("PostCancelAll", request)
+    cancelGetTickersStreamByCount = async (streamNumber: number): Promise<boolean> => {
+        return this.cancelStreamByCount("GetTickersStream", streamNumber)
     }
 
-    postSettle(request: PostSettleRequest): Promise<PostSettleResponse> {
-        return this.wsSocketCall("PostSettle", request)
+    cancelGetTradesStreamByCount = async (streamNumber: number): Promise<boolean> => {
+        return this.cancelStreamByCount("GetTradesStream", streamNumber)
     }
 
-    postReplaceByClientOrderID(request: PostOrderRequest): Promise<PostOrderResponse> {
-        return this.wsSocketCall("postReplaceByClientOrderID", request)
+    cancelGetOrderStatusStreamByCount = async (streamNumber: number): Promise<boolean> => {
+        return this.cancelStreamByCount("GetOrderStatusStream", streamNumber)
     }
 
-    postReplaceOrder(request: PostReplaceOrderRequest): Promise<PostOrderResponse> {
-        return this.wsSocketCall("postReplaceOrder", request)
+    cancelGetRecentBlockhashStreamByCount = async (streamNumber: number): Promise<boolean> => {
+        return this.cancelStreamByCount("GetRecentBlockHashStream", streamNumber)
     }
 
-    getRecentBlockHashStream(request: GetRecentBlockHashRequest): Promise<AsyncGenerator<GetRecentBlockHashResponse>> {
-        return this.wsSocketStreamCall("GetRecentBlockHashStream", request)
+    cancelGetQuotesStreamByCount = async (streamNumber: number): Promise<boolean> => {
+        return this.cancelStreamByCount("GetRecentBlockHashStream", streamNumber)
     }
 
-    // Amm
-    getPrice(request: GetPriceRequest): Promise<GetPriceResponse> {
-        return this.wsSocketCall("GetPrice", request)
+    cancelGetPoolReservesStreamByCount = async (streamNumber: number): Promise<boolean> => {
+        return this.cancelStreamByCount("GetPoolReservesStream", streamNumber)
     }
 
-    getPricesStream(request: GetPricesStreamRequest): Promise<AsyncGenerator<GetPricesStreamResponse>> {
-        return this.wsSocketStreamCall("GetPricesStream", request)
+    cancelAllStreams = async (streamName: string): Promise<Awaited<boolean>[]> => {
+        const retValues: Promise<boolean>[] = []
+
+        const countToSubscriptionID = this.streamToCountMap.get(streamName)
+
+        if (countToSubscriptionID) {
+            countToSubscriptionID.forEach((value: string, key: number) => {
+                this.countToSubscriptionID.delete(key)
+                retValues.push(this.wsConnection.unsubscribe(value))
+            })
+        } else {
+            retValues.push(Promise.reject())
+            return Promise.all(retValues)
```

Comment:
probably should just simplify this to `Promise.reject(new Error("no streams open"))`